---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# scrunchy <img src="man/figures/logo.png" align="right" alt="" width="120" />

[![Travis build status](https://travis-ci.org/hesselberthlab/scrunchy.svg?branch=master)](https://travis-ci.org/hesselberthlab/scrunchy)
[![Codecov test coverage](https://codecov.io/gh/hesselberthlab/scrunchy/branch/master/graph/badge.svg)](https://codecov.io/gh/hesselberthlab/scrunchy?branch=master)
[![Lifecycle badge](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://www.tidyverse.org/lifecycle/#experimental)

scrunchy provides analysis tools for the **S**ingle-**c**ell **r**econstruction of f**unc**tional **h**eterogeneit**y** (**scrunchy**).

Most single-cell analysis experiments provide information about the abundance of molecules in a cell (e.g., mRNAs). We have expanded the capability of single-cell mRNA sequencing experiments to also recover **functional** information---i.e. the activites of enzymes in single cells.

## Installation

Install the R package:

``` r 
# install.packages("remotes")
remotes::install_github("hesselberthlab/scrunchy")
```

# Example

An example data set (`fsce_small`) in scrunchy includes:

+ A single-cell mRNA sequencing experiment (10x Genomics 3prime V2)

+ Counts of DNA repair on synthetic DNA substrates containing a uracil (A:U) and a ribonucleotide (rG:C).

`fsce_small` object now contains mRNA expression (`SCERnaSeq`) and DNA repair activity (`SCEHaircut`) for 250 cells.

```{r example_fsce}
library(scrunchy)

fsce_small
```

These plots illustrate:

1. Two-dimensional embedding of cells defined by UMAP (`umap::umap()`) and colored by k-means clustering (n = 6)

1. Activities on a U:A and riboG DNA hairpins.

```{r example_plots, echo=FALSE, warning=FALSE}
library(ggplot2)

cowplot::plot_grid(
    plot_dims(fsce_tidy, UMAP1, UMAP2, k_cluster, size = 1) +
      ggtitle("mRNA expression"),
    plot_activity(fsce_tidy, Uracil_45, k_cluster) + 
      ggtitle("Uracil excision repair"),
    plot_activity(fsce_tidy, riboG_44, k_cluster) + 
      ggtitle("Ribonucleotide excision repair"),
  nrow = 2,
  axis = "lb",
  align = "hv"
)
```
