
<!-- README.md is generated from README.Rmd. Please edit that file -->

# scrunchy <img src="man/figures/logo.png" align="right" alt="" width="120" />

[![Travis build
status](https://travis-ci.org/hesselberthlab/scrunchy.svg?branch=master)](https://travis-ci.org/hesselberthlab/scrunchy)
[![Codecov test
coverage](https://codecov.io/gh/hesselberthlab/scrunchy/branch/master/graph/badge.svg)](https://codecov.io/gh/hesselberthlab/scrunchy?branch=master)

scrunchy provides a bioinformatic solution for the **S**ingle-**c**ell
**r**esolution of f**unc**tional **h**eterogeneit**y** experiments
(**scrunchy**).

1.  A [Snakemake](https://snakemake.readthedocs.io/en/stable/) pipeline
    for analysis of raw data (`inst/pipeline`).

2.  An R package for analysis of processed data.

## Protocol

A detailed protocol for performing functional measurements of DNA repair
in the context of 10x Genomics single-cell mRNA sequencing experiments
is available at
[protocols.io](https://www.protocols.io/view/single-cell-analysis-of-functional-heterogeneity-i-uhyet7w).

## Installation

Install the R analysis package:

``` r
devtools::install_github("hesselberthlab/scrunchy")
```

Install the Snakemake pipeline:

``` python
nope
```
