---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# scrunchy <img src="man/figures/logo.png" align="right" alt="" width="120" />

[![Travis build status](https://travis-ci.org/hesselberthlab/scrunchy.svg?branch=master)](https://travis-ci.org/hesselberthlab/scrunchy)
[![Codecov test coverage](https://codecov.io/gh/hesselberthlab/scrunchy/branch/master/graph/badge.svg)](https://codecov.io/gh/hesselberthlab/scrunchy?branch=master)
![](https://img.shields.io/badge/lifecycle-experimental-orange.svg)

scrunchy provides analysis tools for the **S**ingle-**c**ell **r**econstruction of f**unc**tional **h**eterogeneit**y** (**scrunchy**).

Most single-cell analysis experiments provide information about the abundance of molecules in a cell (e.g., mRNAs). We have expanded the capability of single-cell mRNA sequencing experiments to also recover **functional** information---i.e. the activites of enzymes in single cells.

## Installation

Install the R package:

``` r 
# install.packages("remotes")
remotes::install_github("hesselberthlab/scrunchy")
```

# Example

An example data set in scrunchy includes:

+ A single-cell mRNA sequencing experiment (10x Genomics 3prime V2)

+ Counts of DNA repair on synthetic DNA substrates containing a uracil (A:U) and a ribonucleotide (rG:C).

```{r example_load, message=FALSE}
library(scrunchy)
library(SingleCellExperiment)

# load mRNA UMI data
exp <- load_csv(scrunchy_data("mrna.csv.gz"))

# load haircut UMI data
fxn <- load_csv(scrunchy_data("haircut.csv.gz"))

fce <- create_fce(exp, fxn)
```

The `fce` object now contains counts for mRNA expression and DNA repair activity for each cell.

```{r example_fce}
fce
```

We can normalize this data, calculate a two-dimensional projection of the mRNA data using `umap::umap()`, and superimpose DNA repair activity.

```{r example_norm}
fce <- normalize_counts(fce)

var_genes <- get_var_genes(fce, n_genes = 5000) 
fce <- calc_pca(fce, n_pcs = 20, genes = var_genes)
fce <- calc_umap(fce, n_dims = 6)

features <- c(
  "Uracil_45",
  "riboG_44",
  "ENSG00000076248",
  "ENSG00000172922"
)

plot_cells(fce, features)
```
