---
title: "Introduction to scrunchy"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction to scrunchy}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r echo=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  out.width = "100%"
)
```

# Create a Functional Single Cell Experiment (fsce)

```{r create_fsce, message=FALSE}
library(scrunchy)

# this same object is available as `fsce_small` in the package.
fsce <- create_fsce(
  list(
    rnaseq = create_sce_rnaseq(scrunchy_data("mrna")),
    haircut = create_sce_haircut(scrunchy_data("haircut"))
  )
)

fsce
```

`fsce` is an object of class `FunctionalSingleCellExperiment`, which can contain data from multiple modalities and does not require that each modality has the same number of cells or features. 

```{r slots}
library(SingleCellExperiment)

# `rnaseq` stores RNA info
fsce[["rnaseq"]]
counts(fsce[["rnaseq"]])[1:5, 1:5]

# `haircut` stores functional data
fsce[["haircut"]]
counts(fsce[["haircut"]])[1:5, 1:5]

# row annotations contain info about each row in the matrix
rowData(fsce[["haircut"]])

# column annotations contain info about each row in the matrix, 
# or about each unique cell present in any of the experiments
colData(fsce)
```

# Plots

`plot_hairpin` produces a coverage plot across each hairpin in the data.

```{r hairpin_plot}
plot_hairpin_coverage(fsce)
```

# Normalization

Normalized data is calculated by the `create_` functions.

RNA data is log-normalized and scaled to total UMI count for each cell. Functional data is normalized using the center log ratio method used for CITE-seq analysis.

Normalized data is stored in the `logcounts` slot of each `SingleCellExperiment` object.

```{r logcounts}
logcounts(fsce[["haircut"]])[1:5, 1:3]
```

# Feature selection and PCA

```{r pca}
# get variable genes
var_genes <- calc_var_features(fsce, n = 1000) 

# calc PCA and embed in 2D
fsce <- calc_pca(fsce, n_pcs = 20, genes = var_genes)
```

# Embed cells in 2D using UMAP or tSNE

```{r 2d}
fsce <- calc_umap(fsce, n_dims = 6)

fsce <- calc_tsne(fsce, n_dims = 6)
```

Cell embeddings are stored in the `reducedDims` slot of the `SingleCellExperiment` and can be accessed as a named list.

```{r rds}
reducedDims(fsce[["rnaseq"]])

reducedDims(fsce[["rnaseq"]])$UMAP[1:4, ]
```

# Tidying data for analysis and plotting

Several `tidy_` functions are provided to convert `fsce` into a tidy formate.

```{r tidy}
library(dplyr)
library(purrr)

fsce_tidy <- purrr::reduce(
  list(
    tidy_dims(fsce_small) %>%
      select(cell_id, starts_with("UMAP"), -experiment),
    tidy_coldata(fsce_small),
    tidy_logcounts(fsce_small[c("Uracil_45", "riboG_44"), , ]) %>%
      select(-experiment)
  ),
  left_join,
  by = "cell_id"
)

fsce_tidy
```

# Visualization

## Two-dimensional embeddings

Features can be visulized on a two-dimensional embedding with the `plot_dims()` function which will plot gene/function assay values.

```{r plot_dims}
cowplot::plot_grid(
  plot_dims(fsce_tidy, UMAP1, UMAP2, Uracil_45, size = 1),
  plot_dims(fsce_tidy, UMAP1, UMAP2, riboG_44, size = 1)
)
```

## Functional activities

Activities can be visualized per-group with `plot_actvitiy()`. These two plots have uracil and ribonucleotide excision activities plotted per cluster (k-means, n = 6).

```{r plot_activity}
cowplot::plot_grid(
  plot_activity(fsce_tidy, Uracil_45, k_cluster, legend = FALSE),
  plot_activity(fsce_tidy, riboG_44, k_cluster, legend = FALSE)
)
```

## Heatmap

These heatmaps show uracil and ribonucleotide excision activities for the U:A and riboG hairpins

```{r heatmap_uracil}
mtx <- logcounts(fsce_small[["haircut"]])

rows <- paste("Uracil", 1:61, sep = "_")
plot_heatmap(mtx, rows, name = "Uracil")
```

```{r heatpmap_riboG}
rows <- paste("riboG", 1:61, sep = "_")
plot_heatmap(mtx, rows, name = "riboG")
```

